### PR TITLE
US 030401 030501

### LINKED ISSUE #39: US 03.04.01
As an administrator, I want to be able to browse events.

### LINKED ISSUE #40: US 03.05.01
As an administrator, I want to be able to browse profiles.

### PR DESCRIPTION
**Pull request by Jameszf**
Updated CRC and UML to match additions.

image
image

# Additions
- Added classes for displaying all users for admins.
- Added classes for displaying all events for admins.
- Instrumented tests for both fragments.
- Custom RecyclerView Adapter abstract classes for creating more adapters in the future.

Screenshot 2024-10-23 180140
Screenshot 2024-10-23 181219

### CODE REVIEW

**Comment by SuperCreativeUniqueUsername** [comment]
My initial comment is to assign yourself to this pr, and assign everyone as a reviewer. Ill look over the code.

File: app/src/main/res/layout/fragment_all_list.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (katzhong) [inline app/src/main/res/layout/fragment_all_list.xml]: could you rename your xml files so that it has the role first? ex. admin_fragment_all_list, etc

File: app/src/main/res/layout/fragment_all_list.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (katzhong) [inline app/src/main/res/layout/fragment_all_list.xml]: NIT: do you think it would be possible to group together then name and dropdown menu icon together in some sort of a box/outline/something so that it is more distinctly a group? something like, maybe?
image

File: app/src/main/java/com/example/pygmyhippo/common/Event.java [hunk 3]
```diff
@@ -31,6 +31,7 @@ public class Event {
     private String organiserID;
```
Comment (katzhong) [inline app/src/main/java/com/example/pygmyhippo/common/Event.java:31]: if you git pull, some of these will be included from my last pr

File: app/src/main/java/com/example/pygmyhippo/common/RecyclerClickListener.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Interface for implementing a click listener for a recycler.
+ * This is generally implemented in the parent fragment of the RecyclerView. You generally need a
+ * reference to the list of data being displayed in the RecyclerView to effectively manipulate it
+ * in onItemClick.
+ *
+ * @see com.example.pygmyhippo.admin.AllEventsFragment
+ */
+public interface RecyclerClickListener {
```
Comment (SuperCreativeUniqueUsername) [inline app/src/main/java/com/example/pygmyhippo/common/RecyclerClickListener.java:11]: I'm curious what this is for?

File: app/src/main/res/navigation/user_mobile_navigation.xml [hunk 5]
```diff
@@ -4,7 +4,21 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/mobile_navigation"
-    app:startDestination="@+id/navigation_home">
+    app:startDestination="@+id/navigation_all_events">
+
+    <fragment
+        android:id="@+id/navigation_all_events"
+        android:name="com.example.pygmyhippo.admin.AllEventsFragment"
```
Comment (SuperCreativeUniqueUsername) [inline app/src/main/res/navigation/user_mobile_navigation.xml:11]: Why are these included in the user navigation and not a new admin nav?

File: app/src/main/java/com/example/pygmyhippo/common/RecyclerClickListener.java (hunk 4 above)
Comment (Jameszf) [reply app/src/main/java/com/example/pygmyhippo/common/RecyclerClickListener.java:11]: For implementing onclick logic for recyclerview items. You implement it in the parent fragment and when on click happens you should have access to the data and position of the item clicked.

File: app/src/main/res/navigation/user_mobile_navigation.xml (hunk 5 above)
Comment (Jameszf) [reply app/src/main/res/navigation/user_mobile_navigation.xml:11]: For testing purposes. I will make a admin nav soon.

**Review by SuperCreativeUniqueUsername — APPROVED** [review]

File: app/src/main/res/layout/fragment_all_list.xml (hunk 2 above)
Comment (Jameszf) [reply app/src/main/res/layout/fragment_all_list.xml]: How is this?
image

File: app/src/main/res/layout/fragment_all_list.xml (hunk 1 above)
Comment (Jameszf) [reply app/src/main/res/layout/fragment_all_list.xml]: renamed

File: app/src/main/res/layout/fragment_all_list.xml (hunk 2 above)
Comment (katzhong) [reply app/src/main/res/layout/fragment_all_list.xml]: like a tiny bit lighter imo 🙏

**Comment by katzhong** [comment]
looks good to me

**Review by katzhong — APPROVED** [review]
